src/percy-node-client.js

```diff
@@ -265,13 +272,72 @@ function snapshot(name, content, opt_breakpoints, opt_enableJs) {
 }
 
 
+/**
+ * Sent another request to the Percy server if the number of tries does not exceed the limit. 
+ * @param {string} buildId Percy Build ID.
+ * @param {number} numRetries The number of get build requests to the server.
+ * @param {function} resolve Promise resolve function.
+ */
+function retry(buildId, numRetries, resolve) {
+  if (numRetries < MAX_RETRIES_WHEN_PROCESSING) {
+      // Retry with recursion with retries incremented
+      return setTimeout(checkBuildStatus, 1000, buildId, numRetries + 1, resolve);
+  } else {
+      handleError('Retries exceeded. Exiting.');
+  }
+}
+
+
+/**
+ * Retrieve the build information from the Percy server, send another request to the server
+ * if the build state is processing or pending. Once the build is finished, check for diffs and
+ * display errors if there are diffs.
+ * @param {string} buildId Percy Build ID.
+ * @param {number} numRetries The number of get build requests to the server.
+ * @param {function} resolve Promise resolve function.
+ */
+async function checkBuildStatus(buildId, numRetries, resolve) {
+  const response = await percyClient.getBuild(buildId);
+  const {body: {data: {attributes}}}  = response;
+  const {state} = attributes;
+  if (state == 'processing' || state == 'pending') {
+      retry(buildId, numRetries, resolve);
+  } else if (state == 'finished'){
+    const totalDiffs = attributes['total-comparisons-diff'];
+      if (totalDiffs) {
+        const url = attributes['web-url'];
+        handleError('percy', `diffs found: ${totalDiffs}. Check ${url}`);
+      } else {
+        logger.log('Hooray! The build is successful with no diffs. \\o/');
+      }
+      resolve();
+  } else if (state == 'failed') {
+    handleError('percy', `build failed: ${attributes['failure-reason']}`);
+    resolve();
+  }
+}
+
+
+/**
+ * Return a promise that gets build information.
+ * @param {string} buildId Percy Build ID.
+ * @return {Promise} Promise object gets resolved when build is finished.
+ */
+function getBuildPromise(buildId) {
+  return new Promise(function(resolve, reject) {
+    checkBuildStatus(buildId, 0, resolve);
+  });
+}
+
+
 /**
  * Finalizes the request to be sent to Percy api which includes all the assets,
  * snapshots, etc.
  * Return this in karma onComplete() phase after all test specs have been run.
+ * @param {boolean} getDiffs Set to true to output build results.
```

Comment:
make it clearer that setting this to true means it's going to wait until the snapshots are analyzed and diffs are generated before the promise is resolved. make it clear that this also slows down the build